main.go

```diff
@@ -396,6 +486,177 @@ func runInPod(name string, cmdToRun string, env []string, timeout int) ([]string
 	return lines[:len(lines)-1], timeout_reached
 }
 
+func selectNodes(step *Step, config Config, subsetPartition map[int][]int) []int{
+	/* Handle OnNode EndNode syntax */
+	if step.OnNode > 0 {
+		if step.Selection != nil {
+			fatal("Two node selection methods on test step")
+		}
+		if step.EndNode == 0 {
+			step.EndNode = step.OnNode
+			selected := make([]int, 1)
+			selected[0] = step.OnNode
+			return selected
+		}
+		return makeRange(step.OnNode, step.EndNode)
+	} else if step.Selection == nil {
+		fatal("No selection method on test step")
+	} else { /* Selection chooses nodes */
+		if (step.Selection.Range == nil && step.Selection.Percent == nil && step.Selection.Subset == nil) {
+			fatal("No selection method on test step")
+		} else if (step.Selection.Range != nil) {
+			return selectNodesRange(step, config)
+		} else if (step.Selection.Percent != nil) {
+			return selectNodesPercent(step, config)
+		} else /* Subset selection */ {
+			return selectNodesSubset(step, subsetPartition)
+		}
+	}
+	/* Should not get here, keeping compiler happy for now*/
+	fatal("Should not reach!")
+	return nil
+}
+
+func selectNodesRange(step *Step, config Config) []int{
+	var selection []int
+	if step.Selection.Range.Order == SEQUENTIAL {
+	  if step.Selection.Range.Start <= 0 || 
```

Comment:
I have the feeling you need a "ValidateSelections()" function which runs before running any tests to make sure things make sense.

Exiting half-way through the tests because of an obvious problem in the selection parameters is not nice. Once you have that you can select your nodes in a much clearer way without this if-then-else nesting.

Also the Go-way tends to bubble up errors, rather than calling `fatal` all around. For example, deciding that these errors are not a fatal and the tests should continue after informing the user needs changing a lot of lines. If these errors were bubbled, it would be matter of changing just the place where `selectNodes` is called.